File: examples/shared/go/interactive/tty.go
package interactive

import (
	"bufio"
	"os"
	"strings"

	"github.com/mattn/go-isatty"
)

// IsTTY checks if stdin is a terminal (TTY).
//
// Returns false if:
//   - Running in a pipe
//   - Running with redirected input
//   - Running in a CI environment without a terminal
func IsTTY() bool {
	return isatty.IsTerminal(os.Stdin.Fd()) || isatty.IsCygwinTerminal(os.Stdin.Fd())
}

// ReadLine reads a single line from stdin, trimmed and lowercased.
// Returns empty string if reading fails.
func ReadLine() string {
	reader := bufio.NewReader(os.Stdin)
	line, err := reader.ReadString('\n')
	if err != nil {
		return ""
	}
	return strings.TrimSpace(strings.ToLower(line))
}

File: examples/shared/rust/src/tty.rs
//! TTY detection and terminal utilities.

use crossterm::tty::IsTty;
use std::io::stdin;

/// Check if stdin is a terminal (TTY).
///
/// Returns false if:
/// - Running in a pipe
/// - Running with redirected input
/// - Running in a CI environment without a terminal
pub fn is_tty() -> bool {
    stdin().is_tty()
}

/// Read a single line from stdin, trimmed.
///
/// Returns None if reading fails or stdin is not available.
pub fn read_line() -> Option<String> {
    let mut input = String::new();
    match std::io::stdin().read_line(&mut input) {
        Ok(_) => Some(input.trim().to_lowercase()),
        Err(_) => None,
    }
}
